table/table.go

```diff
@@ -741,7 +740,64 @@ func (table *Table) InitRoutes(config cfg.Config, meta toml.MetaData) error {
 
 			route, err := route.NewKafkaRoute(routeConfig.Key, routeConfig.Prefix, routeConfig.Substr, routeConfig.Regex, writerConfig, routingMutator)
 			table.AddRoute(route)
+		case "bg_metadata":
+			bgMetadataCfg := routeConfig.BgMetadata
+			if bgMetadataCfg == nil {
+				return fmt.Errorf("error adding route '%s': bg_metadata config is not specified", routeConfig.Key)
+			}
+			if bgMetadataCfg.ShardingFactor == 0 {
+				return fmt.Errorf("error adding route '%s': sharding factor must be specified", routeConfig.Key)
+			}
+			if bgMetadataCfg.FilterSize == 0 {
+				return fmt.Errorf("error adding route '%s': filter size must be specified", routeConfig.Key)
+			}
+			if bgMetadataCfg.FaultTolerance == 0 {
+				return fmt.Errorf("error adding route '%s': fault tolerance percentage must be specified", routeConfig.Key)
+			}
+			if bgMetadataCfg.FaultTolerance < 0 || bgMetadataCfg.FaultTolerance > 1 {
```

Comment:
0 and 1 are invalid values, you need to check for 1 also

Reply:
you're right. done